### PR TITLE
ES6 module import syntax. Fixes #347

### LINKED ISSUE #347: JS error and admin freezes
I am using the latest master from the repo. When I try to load the admin I am getting "Uncaught TypeError: Super expression must either be null or a function, not object" in the console. The admin interface does not load after that.

desktop screenshot

### PR DESCRIPTION
**Pull request by coreygrunewald**
- Fixes ES5 -> ES6 module import syntax.
- Fixes circular dependency issue for `richPromise.jsx`.
- Fixes issue with the Publish dialog when no servers are present in project file.
- Various syntax improvements.

I ran the JS tests locally, all passed. I tried running the Python tests, saw some errors, but they seemed all to originate from dependencies (My knowledge with python and it's tooling is still a bit rough)? I also ran through some manual testing, everything seemed OK, but it would probably be good to get some other hands on it.

One question:

How do the production assets get built when Lektor is deployed to pip / installed from the shell script?

A few recommendations for tasks:
- Add eslint for JS linting.
- Add `prettier` for JS formatting.

### CODE REVIEW

**Comment by tariquesani** [comment]
Thanks for this.

File: lektor/admin/static/js/richPromise.jsx [hunk 1]
```diff
@@ -1,38 +1,35 @@
+import ErrorDialog from './dialogs/errorDialog';
+import dialogSystem from './dialogSystem';
+
 function bringUpDialog(error) {
   // we need to import this here due to circular dependencies
```
Comment (singingwolfboy) [inline lektor/admin/static/js/richPromise.jsx:5]: Can you remove this outdated comment?

**Review by singingwolfboy — APPROVED** [review]
Thanks so much! I just tested this out locally and it seems to work great, although the JS console does seem to pop a few warnings related to using React in deprecated ways. Still, that can be addressed in a separate pull request.

And yes, adding `eslint` would be good -- and I didn't know about `prettier`, but I'll check it out!

File: lektor/admin/static/js/richPromise.jsx (hunk 1 above)
Comment (singingwolfboy) [reply lektor/admin/static/js/richPromise.jsx:5]: Actually, I'll do it post-merge.

**Pull request merged**